m4/ax_check_botan.m4

```diff
@@ -111,6 +111,23 @@ AC_DEFUN([AX_CHECK_BOTAN], [
             AC_MSG_RESULT([no])
             $2
         ])
+
+    AC_MSG_CHECKING([for botan version >= 2.1])
+    AC_COMPILE_IFELSE(
+      [AC_LANG_PROGRAM([[#include <botan/build.h>]], [[
+      #if BOTAN_VERSION_MAJOR >= 2 && BOTAN_VERSION_MINOR >= 1
+      #else
+      #error Botan version 2.1 or newer is required
+      #endif
+      ]])],
+      [
+          AC_MSG_RESULT([ok])
+          $1
+      ], [
+          AC_MSG_RESULT([failed])
+          $2
+      ]
+    )
```

Comment:
2.2 is needed already. The current, unreleased git version of botan sets the version in build.h to 2.2.0, so we can use the same logic, just the value is different.